### PR TITLE
code: Add texture sampling option

### PR DESCRIPTION
**Pull request by GPUCode**
Replaces the nearest neighbour filter that shouldn't have existed in the first place, as it blew up texture sizes for no reason. Forcing nearest neighbour can be useful in some games to avoid blurry graphics, while forcing linear filtering fixes some games like Attack On Titan

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7118)

### CODE REVIEW

File: src/common/settings.h [hunk 1]
```diff
@@ -72,10 +72,15 @@ enum class TextureFilter : u32 {
     None = 0,
     Anime4K = 1,
     Bicubic = 2,
-    NearestNeighbor = 3,
-    ScaleForce = 4,
-    xBRZ = 5,
-    MMPX = 6
+    ScaleForce = 3,
```
Comment (vitor-k) [inline src/common/settings.h:75]: It's not a big deal but we should coordinate a bit when canary tagging or merging this for the minor disruption of users (especially in the cases NearestNeighbor -> Scaleforce and MMPX -> invalid)

Comment (FearlessTobi) [reply src/common/settings.h:75]: Maybe we could just leave the indexes as is?

Comment (GPUCode) [reply src/common/settings.h:75]: Im not a fan of having an index gap in the enum simply for backwards compatibility

**Review by Steveice10 — APPROVED** [review]
LGTM assuming we're fine with the enum shift.

**Comment by GPUCode** [comment]
Avoid merging for a bit, I need to find a better place for this setting

**Comment by Immersion95** [comment]
Unfortunately this setting can't be saved in the general/custom game config https://github.com/citra-emu/citra/issues/7205

**Pull request merged**